### PR TITLE
Cache Refactoring

### PR DESCRIPTION
**Pull request by trickreich**
| Q | A
| --- | ---
| Fixed tickets | --
| License | MIT

#### What's in this PR?

Documentation for Cache Refactoring.

### CODE REVIEW

File: bundles/audience_targeting.rst [hunk 1]
```diff
@@ -58,64 +58,56 @@ And the routes for the website in the corresponding configuration file
 Finally the cache has to be correctly configured. You have the choice between
 the Symfony Cache and Varnish.
 
-For the `Symfony cache`_ the front controller for the website at
-`web/website.php` has to be told to make use of the user context feature. You
-do that by uncommenting the line with the `WebsiteCache` and adding `true` as a
-second parameter:
+For the `Symfony cache`_ the audience targeting cache listener needs to be added.
+This is possible in the constructor of the `WebsiteCache` in `app/WebsiteCache.php`:
```
Comment (danrot) [inline bundles/audience_targeting.rst]: `by adding the constructor`?

File: bundles/audience_targeting.rst [hunk 2]
```diff
@@ -58,64 +58,56 @@ And the routes for the website in the corresponding configuration file
...
+    }
 
 If you want to use the more powerful `Varnish`_ instead, you have to install it
-on your machine and configure it using a VCL. The following example can be used
-for that, however, it requires the header module, which comes with the
-`varnish-modules`_ package.
+on your machine and configure it using a VCL.
 
-.. code-block:: c
+The following will add full caching support inclusive audience targeting for Sulu:
```
Comment (danrot) [inline bundles/audience_targeting.rst]: `including`

File: bundles/audience_targeting.rst [hunk 3]
```diff
@@ -58,64 +58,56 @@ And the routes for the website in the corresponding configuration file
...
 
+.. code-block:: varnish4
+
+    # /etc/varnish/default.vcl
     vcl 4.0;
 
-    import header;
+    include "<PATH-TO-SULU>/src/Sulu/Bundle/HttpCacheBundle/Resources/varnish/sulu.vcl";
+    include "<PATH-TO-SULU>/src/Sulu/Bundle/AudienceTargetingBundle/Resources/varnish/sulu.vcl";
```
Comment (danrot) [inline bundles/audience_targeting.rst]: I think I would write `<PATH-TO-SULU-MINIMAL>/vendor/sulu/sulu` instead. I guess many people don't know that there is another package and where it is located.

File: bundles/http_cache.rst [hunk 4]
```diff
@@ -1,104 +1,86 @@
 HttpCacheBundle
 ===============
 
-The SuluHttpCache bundle provides tight integration between Sulu and HTTP caching proxies.
+The SuluHttpCache bundle provides integration between Sulu <--> FOSHttpCacheBundle <--> HTTP caching proxies.
```
Comment (danrot) [inline bundles/http_cache.rst]: `The SuluHttpCache bundle provides integration between Sulu and HTTP caching proxies using the FOSHttpCacheBundle`

Also include a link to the `FOSHTTPCacheBundle`.

File: cookbook/caching-with-varnish.rst [hunk 5]
```diff
@@ -133,9 +133,7 @@ The following will add full caching support for Sulu:
     # /etc/varnish/default.vcl
     vcl 4.0;
 
-    C{
-        #include <stdlib.h>
-    }C
+    include "<PATH-TO-SULU>/src/Sulu/Bundle/HttpCacheBundle/Resources/varnish/sulu.vcl";
```
Comment (danrot) [inline cookbook/caching-with-varnish.rst]: Same as above

**Comment by danrot** [comment]
@trickreich The build is failing.

**Comment by trickreich** [comment]
@danrot Build works now

**Comment by danrot** [comment]
@trickreich There is a conflict for some reason.

**Pull request merged**